test/test_generators.py

```diff
@@ -103,28 +298,38 @@ def test_full_generator_windows(tmpdir):
     st_window_shape, min_size=1, max_size=10, unique=True)
 
 
-def create_step_and_image_strategy(limit):
-
+def create_step_and_image_strategy(args):
+    limit, dtype = args
     step_size = st.tuples(
         st.integers(min_value=1, max_value=limit[0]),
         st.integers(min_value=1, max_value=limit[1]),
     )
 
+    if np.issubdtype(dtype, np.floating):
+        elements = st.floats(min_value=0, max_value=1)
+    else:
+        elements = st.integers(min_value=np.iinfo(dtype).min,
+                               max_value=np.iinfo(dtype).max)
+
     image_array = arrays(
-        dtype=st_rasterio_dtypes,
+        dtype=dtype,
         shape=st.tuples(
             st.just(len(BANDS['quickbird'])),
             st.integers(min_value=limit[0], max_value=10),
             st.integers(min_value=limit[1], max_value=10),
         ),
+        elements=elements
     )
 
     return st.tuples(step_size, image_array)
 
 
 st_step_and_image = st.tuples(
-    st.integers(min_value=1, max_value=10),
-    st.integers(min_value=1, max_value=10),
+    st.tuples(
+        st.integers(min_value=1, max_value=10),
+        st.integers(min_value=1, max_value=10)
+    ),
+    st_rasterio_dtypes
```

Comment:
```suggestion
    st_rasterio_dtypes,
```

Reply:
The preferred style is to also add a comma after the last element in a multi-line list/dict/etc., see https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas